blockdev/src/blockdev.rs

```diff
@@ -240,6 +303,123 @@ impl Drop for LoopbackDevice {
+pub fn run_loopback_cleanup_helper(device_path: &str, parent_pid: u32) -> Result<()> {
+    use std::os::unix::process::CommandExt;
+    use std::process::Command;
+
+    // Check if we're running as a cleanup helper
+    if std::env::var("BOOTC_LOOPBACK_CLEANUP_HELPER").is_err() {
+        anyhow::bail!("This function should only be called as a cleanup helper");
+    }
+
+    // Close stdin, stdout, stderr and redirect to /dev/null
+    let null_fd = std::fs::File::open("/dev/null")?;
+    let null_fd = null_fd.as_raw_fd();
+    unsafe {
+        libc::dup2(null_fd, 0);
+        libc::dup2(null_fd, 1);
+        libc::dup2(null_fd, 2);
+    }
+    
+    // Set up death signal notification - we want to be notified when parent dies
+    unsafe {
+        if libc::prctl(libc::PR_SET_PDEATHSIG, libc::SIGUSR1) != 0 {
+            std::process::exit(1);
+        }
+    }
+
+    // Mask most signals to avoid being killed accidentally
+    // But leave SIGUSR1 unmasked so we can receive the death notification
+    unsafe {
+        let mut sigset: libc::sigset_t = std::mem::zeroed();
+        libc::sigfillset(&mut sigset);
+        // Don't mask SIGKILL, SIGSTOP (can't be masked anyway), or our death signal
+        libc::sigdelset(&mut sigset, libc::SIGKILL);
+        libc::sigdelset(&mut sigset, libc::SIGSTOP);
+        libc::sigdelset(&mut sigset, libc::SIGUSR1); // We'll use SIGUSR1 as our death signal
+
+        if libc::pthread_sigmask(libc::SIG_SETMASK, &sigset, std::ptr::null_mut()) != 0 {
+            let err = std::io::Error::last_os_error();
+            tracing::error!("pthread_sigmask failed: {}", err);
+            std::process::exit(1);
+        }
+    }
+
+    // Wait for death signal or normal termination
+    let mut siginfo: libc::siginfo_t = unsafe { std::mem::zeroed() };
+    let sigset = {
+        let mut sigset: libc::sigset_t = unsafe { std::mem::zeroed() };
+        unsafe {
+            libc::sigemptyset(&mut sigset);
+            libc::sigaddset(&mut sigset, libc::SIGUSR1);
+            libc::sigaddset(&mut sigset, libc::SIGTERM); // Also listen for SIGTERM (normal cleanup)
+        }
+        sigset
+    };
+
+    // Wait for a signal
+    let result = unsafe {
+        let result = libc::sigwaitinfo(&sigset, &mut siginfo);
+        if result == -1 {
+            let err = std::io::Error::last_os_error();
+            tracing::error!("sigwaitinfo failed: {}", err);
+            std::process::exit(1);
+        }
+        result
+    };
+
+    if result > 0 {
+        if siginfo.si_signo == libc::SIGUSR1 {
+            // Parent died unexpectedly, clean up the loopback device
+            let status = std::process::Command::new("losetup")
+                .args(["-d", device_path])
+                .status();
+
+            match status {
+                Ok(exit_status) if exit_status.success() => {
+                    // Write to stderr since we closed stdout
```

Comment:
Mmm my vote here is probably to not inherit stderr at all; if we write to stderr then we have the possibility to intermix the child process writes with the parent's.

One option is to explicitly log to the systemd journal.

I guess speaking of systemd...a whole possibility I hadn't considered until just now is that we fork off via `systemd-run`. That would have some nice advantages but will be trickier to get right the lifecycle binding, so let's leave that for the future.

(There's this whole giant topic in bootc overall defaulting to running via systemd in some cases, which would similarly have a lot of advantages but be a big nontrivial change)